storage/jellyfish_merkle/src/restore/restore_test.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    mock_tree_store::MockTreeStore, restore::JellyfishMerkleRestore, JellyfishMerkleTree,
+    TreeReader,
+};
+use crypto::HashValue;
+use proptest::{collection::btree_map, prelude::*};
+use std::collections::BTreeMap;
+use types::{account_state_blob::AccountStateBlob, transaction::Version};
+
+proptest! {
+    #![proptest_config(ProptestConfig::with_cases(10))]
+
+    #[test]
+    fn test_restore_without_interruption(
+        btree in btree_map(any::<HashValue>(), any::<AccountStateBlob>(), 1..1000),
+    ) {
+        let version = btree.len() as Version - 1;
```

Comment:
Slightly prefer `btree.len() - 1 as Version` as then arithmetic is done on usize instead of Version (assuming this doesn't hit some silly clippy rule).  Up to you if you want to change; either way is fine.